src/utils/processorUtils.ts

```diff
@@ -0,0 +1,229 @@
+    )
+    .map((filter) => {
+      if (filter.value && isCommaSeparatedFilterType(filter)) {
+        const multipleValuesMapper = (item: string): number | string =>
+          filter.type === FilterType.NUMBER_IN
+            ? parseFloat(item.trim())
+            : item.trim();
+
+        const multipleValuesPredicate =
+          filter.type === FilterType.NUMBER_IN
+            ? (item: number | string): boolean =>
+                item !== null && !isNaN(item as number)
+            : String;
+
+        const values = filter.value
+          .split(",")
+          .map(multipleValuesMapper)
+          .filter(multipleValuesPredicate);
+
+        return {
+          key: filter.key,
+          type: filter.type,
+          values,
+        };
+      }
+      return filter;
+    })
+    .filter(
+      (item: Partial<EventFilter>) =>
+        item.value || (item.values && item.values.length > 0)
+    );
+
+/**
+ * Cleanup and prepare processor form data before sending it to an API request
+ * @param formData Data coming from the processor form
+ * @param isExistingProcessor Flag indicating if it's a new processor or an update
+ * @param schema The json-schema of the action/source configuration used in the processor
+ */
+export const prepareRequest = (
+  formData: ProcessorFormData,
+  isExistingProcessor: boolean,
+  schema: object
+): ProcessorRequest => {
+  const requestData: ProcessorRequest = { name: formData.name };
+  if (formData.type === ProcessorType.Sink) {
+    requestData.action =
+      isExistingProcessor && formData.action && schema
+        ? {
+            type: formData.action.type,
+            parameters: prepareConfigParameters(
+              formData.action.parameters,
+              schema,
+              "write"
+            ),
+          }
+        : formData.action;
+  } else {
+    requestData.source =
+      isExistingProcessor && formData.source && schema
+        ? {
+            type: formData.source.type,
+            parameters: prepareConfigParameters(
+              formData.source.parameters,
+              schema,
+              "write"
+            ),
+          }
+        : formData.source;
+  }
+  if (formData.filters && formData.filters.length > 0) {
+    const filtersData = prepareFilters(formData.filters) as EventFilter[];
+    if (filtersData.length > 0) {
+      requestData.filters =
+        filtersData as unknown as ProcessorRequest["filters"];
+    }
+  }
+  if (
+    formData.transformationTemplate &&
+    formData.transformationTemplate.trim().length > 0
+  ) {
+    requestData.transformationTemplate = formData.transformationTemplate.trim();
+  }
+  return requestData;
+};
+
+/**
+ * Retrieve the definition of a specific property inside a json schema
+ * @param propertyName Name of the property
+ * @param schema Json schema object
+ * @param [parentSchema] Parent schema object used when drilling down nested fields
+ */
+export const getSchemaDefinition = (
```

Comment:
Retrieving the definition of a property inside a schema. Used when handling secret parameters to decide if the parameter is defined as secret or not.